### PR TITLE
Fix equality of empty variants (with and without attributes)

### PR DESCRIPTION
**Pull request by siddhukrs**
Empty variants, with and without attributes, were failing to pass an equality check because our code wasn't handling nullptrs as data values. 

Fixing those cases and adding tests.

### CODE REVIEW

File: source/core/DualTypeVisitor.cpp [hunk 1]
```diff
@@ -61,6 +61,9 @@ namespace Vireo
                         success = ArrayCompatible(typeRefX, typeRefY, operation);
                     }
                     break;
+                case kEncoding_None:
```
Comment (siddhukrs) [inline source/core/DualTypeVisitor.cpp]: The `underlyingType` on empty variant constants is of `kEncoding_None`. Treating them as equals in the `TypesCompatible` check and `Apply` cases.

File: source/core/Variants.cpp [hunk 2]
```diff
@@ -439,7 +439,8 @@ bool VariantsAreEqual(VariantTypeRef variantX, VariantTypeRef variantY)
 {
     DualTypeVisitor visitor;
     DualTypeEqual dualTypeEqual;
-    return visitor.Visit(variantX, variantX->Begin(kPARead), variantY, variantY->Begin(kPARead), &dualTypeEqual);
+    return (!variantX && !variantY)  // Empty variant constant with no attributes are equal
```
Comment (siddhukrs) [inline source/core/Variants.cpp]: Empty variant constants with no attributes are plain nullptrs. Handling that special case here. 

However, if they do go through a `SetVariantAttribute` instruction they are initialized and have attribute maps and go through the visitor.

File: test-it/ExpectedResults/TestVI_VariantEquality_AttributeComparison.vtr [hunk 3]
```diff
@@ -1,3 +1,6 @@
+TestResult_EmptyVariantConstantWithEqualDataAndUnequalAttributeNameAndValue true
```
Comment (siddhukrs) [inline test-it/ExpectedResults/TestVI_VariantEquality_AttributeComparison.vtr]: emptyvariants

Comment (siddhukrs) [reply test-it/ExpectedResults/TestVI_VariantEquality_AttributeComparison.vtr]: I also added a test that compares empty variant constant vs variant created using ToVariant.

Comment (cglzaguilar) [reply test-it/ExpectedResults/TestVI_VariantEquality_AttributeComparison.vtr]: Do we have tests that cover empty variant constant vs variant with data?

File: source/core/Variants.cpp [hunk 4]
```diff
@@ -439,6 +439,11 @@ bool VariantsAreEqual(VariantTypeRef variantX, VariantTypeRef variantY)
 {
     DualTypeVisitor visitor;
     DualTypeEqual dualTypeEqual;
+    if (!variantX && !variantY) {
```
Comment (cglzaguilar) [inline source/core/Variants.cpp]: I think we need to move this check inside the visitor or the operation. For inner variants this case may present again.

File: source/core/DualTypeVisitor.cpp (hunk 1 above)
Comment (cglzaguilar) [reply source/core/DualTypeVisitor.cpp]: I think we need to verify the encoding of typeRefY is also kEncoding_None and return true in that case.

**Comment by siddhukrs** [comment]
Will re-open this PR after this other one is in: https://github.com/ni/VireoSDK/pull/610
